### ISSUE TITLE
RxUI.Events + ElementCompositionPreview in same class leads to compilation error

### ISSUE THREAD

**Issue by jasonwurzel** [reactiveui/ReactiveUI#1668]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

Having a UWP project. When having the following namespace import at the top (e.g. neccessary for ElementCompositionPreview class) 
```csharp
using Windows.UI.Xaml.Hosting;
```
and using ReactiveUI.Events at the same time in the same file there is an compile time error stating that 
```
CS7069    Reference to type 'DesignerAppManager' claims it is defined in 'Windows', but it could not be found
```

The cause is that the using statement imports the Events extension for DesignerAppManager, but this class itself is defined in "Windows Desktop Extension SDK". 
If I reference the said SDK via Project References => Universal Windows => Extensions => "Windows Desktop Extensions for the UWP" the error vanishes (because DesignerAppManager can be found)

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

1. create any uwp app project targeting at least SDK 16299
2. reference RxUI.Events
3. use ElementCompositionPreview in any class
4. use Events extension method in same class
5. the mentioned error is present (uiElement.Events()... is underlined red)

(repro: https://github.com/jasonwurzel/rxui-events-problem )

**What is the expected behavior?**

not having to reference Windows Desktop Extension SDK when it is not needed.

**What is the motivation / use case for changing the behavior?**

using ElementCompositionPreview class (Windows.UI.Xaml.Hosting namespace) in the same file as RxUI.Events extension.

**Which versions of ReactiveUI, and which platform / OS are affected by this issue? Did this work in previous versions of ReativeUI? Please also test with the [latest stable](https://www.nuget.org/profiles/reactiveui) and [development snapshot](https://reactiveui.net/docs/getting-started/installation/nuget-packages/development-package-feed)**

RxUI 8.2.1
Windows 10 (17134)

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**

The Events package should only contain extension methods for classes present in a naked UWP app project. How/Why are there generated extension methods for types present in the Windows Desktop Extension SDK? What should we ideally do about it? An extra nuget package for users who reference the WDE SDK? (maybe there are other cases for other nuget packages as well?)

**Comment by jasonwurzel** [comment]
Hi, here's a quick repro of the problem: 
https://github.com/jasonwurzel/rxui-events-problem

**Comment by glennawatson** [comment]
Did you want to investigate it @arggrande ?

One thing to note, nuget references are lazy, so it won't be loaded until you actually use something in that extensions SDK.

**Comment by harvinders** [comment]
The work around is to create a partial class and keep the code in separate classes.

**Comment by jasonwurzel** [comment]
Hi @harvinders, thank you for the work around, that's a very good idea!

**Comment by jasonwurzel** [comment]
@harvinders , been playing around with this workaround and sadly it does work only in certain circumstances. 
As the extensions live in the same namespace as the classes/controls the extensions are for, I can't circumvent the error in the following case: 

```csharp
using Windows.UI.Xaml.Controls;

namespace MyControls
{
    public sealed partial class MirandaShell
    {
        public NavigationViewEvents ShellNavigationViewEvents => ShellNavigationView.Events();
    }
}
```

I get the following errors: 

![image](https://user-images.githubusercontent.com/1893675/64599303-6b0dfc80-d3b9-11e9-912a-816c7e4661d5.png)

**Comment by glennawatson** [comment]
Would a solution to your problem be potentially putting them in their own namespace?

Eg maybe put some sort of prefix for the events.

**Comment by harvinders** [comment]
@jasonwurzel Could you please try version 9.19.2. I have also seen SwipeItem issue in my project but when using 9.19.2 I don't see it. I suspect something got broken after that.

**Comment by jasonwurzel** [comment]
@glennawatson 
this is a good question I unfortunately have no answer for. I updated the [repro I built](https://github.com/jasonwurzel/rxui-events-problem) to use the newest version of ReactiveUI.Events. Interestingly, now the error has changed (same as in my screenshot above). In contrast to my initial bug description, it has nothing to do with referencing ElementCompositionPreview or a `using Windows.UI.Xaml.Hosting` - see my updated repro. I just use `button.Events()....` and get the errors.
Sadly, I'm a bit at a loss here. I guess putting the extensions in their own prefixed namespace should be a workaround, because then the `using Windows.UI.Xaml.Controls;` can be changed to `using RxEvents.Windows.UI.Xaml.Controls;` or sth. similar. 
Yet I guess the base problem remains (my guess because I don't know how nowadays the pharmacist generates the files): ReactiveUI.Events generates Events_uwp.cs out of some reference/contract that is different from what is present when we create a naked new UWP Project.

@harvinders 
Interesting! In 9.19.2 the TargetFramework for UWP got changed from 10.0.17763 down to 10.0.16299. I'll try that, yet my guess is the problem will shift but raise it's head on some other occasion (see above the difference from using RxUI 8 in my original repro to using the current RxUI 10.1.6. Error has changed but same pattern.)
